org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java

```diff
@@ -138,11 +138,12 @@ private void updateMarkers(PublishDiagnosticsParams diagnostics, IResource resou
 		}
 		IWorkspaceRunnable runnable = monitor -> {
 			if (resource.exists()) {
+				IDocument document = LSPEclipseUtils.getDocument(resource);
```

Comment:
The call LSPEclipseUtils.getDocument(resource) has an extra code path which was not present in this file. I do not think that is a problem, but if it is a problem I can rework the PR avoid the extra code path.